### ISSUE TITLE
 Error: Unknown database type postgres:

### ISSUE THREAD

**Issue by emolr** [prisma/prisma2#599]
Hi, My project works fine with prisma@2.0.0-preview-11 using postgres.

But running:
```
npx prisma2@2.0.0-preview-12 lift up
```

outputs: 
```
Error: Unknown database type postgres:
```

**Comment by divyenduz** [comment]
Please try `postgresql`, this change was made a while ago though. I am not sure how it only emerged now.

**Comment by divyenduz** [comment]
Please share your full schema as well, if possible

**Comment by emolr** [comment]
I will do when possible @divyenduz :) 

I can elaborate a bit on the error. 

The database it refers to as type postgres, is actually the beginning of my postgres url in the url property of my datasource.

If I change my url to eg. notvalidurl://username:password@host:5432

It will print ‘ Error: Unknown database type notvalidurl:’

Hope it makes more sense

**Comment by divyenduz** [comment]
Thanks, I understand, I think this was fixed to "postgresql" a few releases ago, you faced this between 11 and 12 confused me. We will try it on both and figure what caused this. 

Please note that supporting only `postgresql` is the intended behavior. An error for `postgres://..` is expected.

**Comment by divyenduz** [comment]
I can confirm that the following works in Preview 11 but does not work in Preview 12

```
datasource db {
  provider = "postgresql"
  url      = "postgres://root:prisma@localhost:5432/postgres?schema=p2-pg"
}
```

I can also confirm that we **need** to support this form of connection string because that is what Heroku provides. Marking this as a bug. 

P.S. This should have been handled in the release notes as well 🙏

**Comment by emolr** [comment]
Perfect :) I updated my code to use `postgresql://` and i can confirm that it works :)

**Comment by tomhoule** [comment]
For the Lift part:

prisma2 PR: https://github.com/prisma/prisma2/pull/751
prisma-engine PR: https://github.com/prisma/prisma-engine/pull/101

**Comment by divyenduz** [comment]
A note for our users: 

This issue tracks releasing alpha when a PR into `prisma-engine` is merged: https://github.com/prisma/prisma2/issues/658

Till that is closed, we manually close issues that rely on Prisma Engine. 

tl;dr; this is not released yet but will be soon.

**Comment by tomhoule** [comment]
@divyenduz  can I close this issue?

**Comment by divyenduz** [comment]
This is available in the latest alpha (`npm install -g prisma2@alpha`) and preview (`npm install -g prisma2`).

**Comment by iRoachie** [comment]
@divyenduz I can't seem to get this to work at all. Event after updating -g and local version of prisma. Anything I am missing?

**Comment by pantharshit00** [comment]
Can you tell me the output of `prisma2 -v`?

**Comment by iRoachie** [comment]
prisma2@2.0.0-preview015, binary version: 20b6dc13949cccccfef5be07c0be7a3d7c858abe

**Comment by iRoachie** [comment]
> Also, make sure to prisma2 dev in order to update your photon client and database.

This throws the error as well:
```
Error: Unknown database type postgres:
```

**Comment by pantharshit00** [comment]
@iRoachie Can you please make a new issue about this with a reproduction 🙏 ?

**Comment by iRoachie** [comment]
Sure @pantharshit00 https://github.com/iRoachie/prisma2-postgres-bug. Not sure if making a new issue makes sense since it's the exact same problem (nothing changed).

The example is the starter template anyway. The bug is very present without any configuration right out of the box.

**Comment by pantharshit00** [comment]
I can confirm this now, looks like we are not accepting `postgres://` url schema when using an environment variable.

The easy workaround is to use `postgresql://` instead of `postgres://` in the url schema.

**Comment by iRoachie** [comment]
This isn't really possible for some environments. 

For example, hosting on Heroku, Postgres changes the URL frequently. When the URL changes, they automatically restart your app with the new URL. To access the db you have to use the environment variable they provide (which unfortunately is `postgres://`). In this scenario, I can't change the URL.

**Comment by pantharshit00** [comment]
That is why I have marked this as a bug and we will fix this :)

**Comment by tomhoule** [comment]
I investigated this for a bit, and it looks like the error comes from inside the lift typescript code:  https://github.com/prisma/prisma2/blob/17567a92bc8f66bac335e19f9cafa772b0567982/cli/introspection/src/convertCredentials.ts#L102

It's fixed apparently, maybe we should try again with preview 16.

**Comment by tomhoule** [comment]
I can also confirm that it happens in the latest alpha.

**Comment by timsuchanek** [comment]
Internal Note: Dropping this for now, as the Rust implementation is replacing the TypeScript implementation within the next weeks.

**Comment by iRoachie** [comment]
@pantharshit00 can I remove the repo now?

**Comment by pantharshit00** [comment]
Yes, you can remove the repo now as we have identified the cause.

**Comment by timsuchanek** [comment]
Thanks a lot for reporting 🙏
This issue is fixed in the latest alpha version of `prisma2`.
You can try it out with `npm i -g prisma2@alpha`.

In case it’s not fixed for you - please let us know and we’ll reopen this issue!
